### PR TITLE
Distributor: Enable automatically generated OTel metric name suffixes

### LINKED ISSUE #5967: OTLP ingest expose addMetricSuffixes as a configuration option
Follow up to https://github.com/grafana/mimir/pull/5693. OTLP ingest expose addMetricSuffixes as a configuration option.

https://github.com/grafana/mimir/pull/5693/files#diff-eb47b1b5619b45185602515d8d5274a06451058d43aaa9c85c9fc1a7222635c3R184

### PR DESCRIPTION
**Pull request by aknuds1**
#### What this PR does
In the distributor's OTel endpoint, support enabling of automatically generated metric name suffixes through the flag `-distributor.otel-metric-suffixes-enabled`.

TODO:

- [x] Fix integration test
- [x] Test manually

#### Which issue(s) this PR fixes or relates to

Fixes #5967.

#### Checklist

- [x] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/distributor/otel.go [hunk 1]
```diff
@@ -203,7 +213,7 @@ func otelMetricsToMetadata(md pmetric.Metrics) []*mimirpb.MetricMetadata {
 				metric := scopeMetrics.Metrics().At(k)
 				entry := mimirpb.MetricMetadata{
 					Type:             otelMetricTypeToMimirMetricType(metric),
-					MetricFamilyName: prometheustranslator.BuildCompliantName(metric, "", true), // TODO expose addMetricSuffixes in configuration (https://github.com/grafana/mimir/issues/5967)
+					MetricFamilyName: prometheustranslator.BuildCompliantName(metric, "", addSuffixes),
```
Comment (jmichalek132) [inline pkg/distributor/otel.go:216]: So, when setting the addSuffixes to true or false it needs to be set here too
https://github.com/grafana/mimir/blob/main/pkg/distributor/otel.go#L220
The `prometheus.Settings{}` passed there has a bool for it too.
```
type Settings struct {
	Namespace           string
	ExternalLabels      map[string]string
	DisableTargetInfo   bool
	ExportCreatedMetric bool
	AddMetricSuffixes   bool
}
```
If both are not set to true or false, the metadata won't show up properly in Grafana, becaus the metric name in TSDB won't match the metric name in Metadata storage if the suffix is injected or not injected in both places.

Comment (aknuds1) [reply pkg/distributor/otel.go:216]: Thank you!

Comment (aknuds1) [reply pkg/distributor/otel.go:216]: Fixed.

**Review by jmichalek132 — APPROVED** [review]

File: integration/otlp_ingestion_test.go [hunk 2]
```diff
@@ -49,13 +67,22 @@ func TestOTLPIngestion(t *testing.T) {
 	c, err := e2emimir.NewClient(mimir.HTTPEndpoint(), mimir.HTTPEndpoint(), "", "", "user-1")
 	require.NoError(t, err)
 
+	sfx := ""
+	if enableSuffixes {
+		sfx = "_bytes"
```
[the remark this answers is not in the document]
Comment (aknuds1) [reply integration/otlp_ingestion_test.go:72]: That means there will always be the suffix of `_` at least.

**Pull request merged**